From: Mykola Golub <mgolub@mirantis.com>
Date: Tue, 29 Nov 2016 11:40:14 +0200
Subject: rbd-mirror: split-brain issues should be clearly visible in mirror
 status

Fixed: http://tracker.ceph.com/issues/16991
Signed-off-by: Mykola Golub <mgolub@mirantis.com>
(cherry picked from commit cccca67d5f898c43207a19a6e029a1abb86efbcb)
(cherry picked from commit e34a403df5e247b0937abee617758e0838fc0c8e)

Resolves: rhbz#1424687
---
 src/tools/rbd_mirror/ImageReplayer.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/tools/rbd_mirror/ImageReplayer.cc b/src/tools/rbd_mirror/ImageReplayer.cc
index 3e9e5f8..b1e1e6b 100644
--- a/src/tools/rbd_mirror/ImageReplayer.cc
+++ b/src/tools/rbd_mirror/ImageReplayer.cc
@@ -428,6 +428,9 @@ void ImageReplayer<I>::handle_bootstrap(int r) {
     dout(5) << "remote image is non-primary or local image is primary" << dendl;
     on_start_fail(0, "remote image is non-primary or local image is primary");
     return;
+  } else if (r == -EEXIST) {
+    on_start_fail(r, "split-brain detected");
+    return;
   } else if (r < 0) {
     on_start_fail(r, "error bootstrapping replay");
     return;
